controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentUpgrader.java

```diff
@@ -32,16 +33,20 @@ public DeploymentUpgrader(Controller controller, Duration interval) {
     protected double maintain() {
         AtomicInteger attempts = new AtomicInteger();
         AtomicInteger failures = new AtomicInteger();
-        Versions target = new Versions(controller().readSystemVersion(), ApplicationVersion.unknown, Optional.empty(), Optional.empty());
+        Version systemVersion = controller().readSystemVersion();
+
         for (Application application : controller().applications().readable())
             for (Instance instance : application.instances().values())
                 for (Deployment deployment : instance.deployments().values())
                     try {
                         attempts.incrementAndGet();
                         JobId job = new JobId(instance.id(), JobType.from(controller().system(), deployment.zone()).get());
                         if ( ! deployment.zone().environment().isManuallyDeployed()) continue;
+
+                        Run last = controller().jobController().last(job).get();
+                        Versions target = new Versions(systemVersion, last.versions().targetApplication(), Optional.empty(), Optional.empty());
                         if ( ! deployment.version().isBefore(target.targetPlatform())) continue;
-                        if (   controller().clock().instant().isBefore(controller().jobController().last(job).get().start().plus(Duration.ofDays(1)))) continue;
+                        if (   controller().clock().instant().isBefore(last.start().plus(Duration.ofDays(1)))) continue;
```

Comment:
Hmm, perhaps we should remove this, for those who deploy every day to dev, so their deployments can be upgraded when they sleep, too?

Reply:
Too scared to introduce more changes in this PR :scream: